### PR TITLE
Sync with pokeemerald

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Ready for review. I'll leave it here for now and I'll continue in a new PR once this one is merged.

**Comment by AsparagusEduardo** [comment]
*Now* it's ready for review.

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -843,31 +793,35 @@ static const struct PickupItem sPickupItems[] =
...
 {
-    20, 15, 10, 15 // Ultra, Great, Poke, Safari
+    [ITEM_ULTRA_BALL - ITEM_ULTRA_BALL]  = 20,
+    [ITEM_GREAT_BALL - ITEM_ULTRA_BALL]  = 15,
+    [ITEM_POKE_BALL - ITEM_ULTRA_BALL]   = 10,
+    [ITEM_SAFARI_BALL - ITEM_ULTRA_BALL] = 15
 };
 
-// not used
+// Not used. Maybe related to gBattlePalaceNatureToMoveGroupLikelihood from Emerald?
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:817]: Seems unlikely, that table is much larger and Emerald came later in development

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -1111,95 +1084,104 @@ static void atk01_accuracycheck(void)
...
+
         // final calculation
         if ((Random() % 100 + 1) > calc)
         {
             gMoveResultFlags |= MOVE_RESULT_MISSED;
-            if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE 
-             && (gBattleMoves[move].target == MOVE_TARGET_BOTH || gBattleMoves[move].target == MOVE_TARGET_FOES_AND_ALLY))
-                gBattleCommunication[6] = 2;
+            if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE &&
+                (gBattleMoves[move].target == MOVE_TARGET_BOTH || gBattleMoves[move].target == MOVE_TARGET_FOES_AND_ALLY))
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:1098]: The original format is more consistent with the rest of the repo.

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -2566,37 +2600,38 @@ void SetMoveEffect(bool8 primary, u8 certain)
...
                     if (GetBattlerSide(gBattlerAttacker) == B_SIDE_OPPONENT
-                     && !(gBattleTypeFlags &
-                          (BATTLE_TYPE_EREADER_TRAINER
-                          | BATTLE_TYPE_BATTLE_TOWER
-                          | BATTLE_TYPE_LINK))
+                        && !(gBattleTypeFlags &
+                             (BATTLE_TYPE_EREADER_TRAINER
+                              | BATTLE_TYPE_BATTLE_TOWER
+                              | BATTLE_TYPE_LINK))
                      && gTrainerBattleOpponent_A != 0x400)
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:2623]: Inconsistent alignment. Also these `0x400`s are `TRAINER_SECRET_BASE`

File: src/battle_script_commands.c [hunk 4]
```diff
@@ -3305,72 +3380,91 @@ static void atk23_getexp(void)
     }
 }
 
-static void atk24(void)
+// For battles that aren't BATTLE_TYPE_LINK or BATTLE_TYPE_RECORDED_LINK, the only thing this
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:3383]: `BATTLE_TYPE_RECORDED_LINK` isn't relevant here in pokefirered

File: src/battle_script_commands.c [hunk 5]
```diff
@@ -6521,61 +6790,74 @@ static u8 ChangeStatBuffs(s8 statValue, u8 statId, u8 flags, const u8 *BS_ptr)
...
         }
-        gBattleTextBuff2[index++] = B_BUFF_STRING;
-        gBattleTextBuff2[index++] = STRINGID_STATROSE;
-        gBattleTextBuff2[index++] = STRINGID_STATROSE >> 8;
+        gBattleTextBuff2[index] = B_BUFF_STRING;
+        index++;
+        gBattleTextBuff2[index] = STRINGID_STATROSE;
+        index++;
+        gBattleTextBuff2[index] = STRINGID_STATROSE >> 8;
+        index++;
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:6798]: This is worse

File: src/battle_script_commands.c [hunk 6]
```diff
@@ -6651,7 +6935,7 @@ static void atk8F_forcerandomswitch(void)
         else
         {
             valid = 0;
-            for (i = 0; i < 6; ++i)
+            for (i = 0; i < 6; i++)
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:6938]: `PARTY_SIZE`. This function (here and in pokeemerald) is missing a bunch of `PARTY_SIZE` and `PARTY_SIZE / 2`

File: src/battle_script_commands.c [hunk 7]
```diff
@@ -7271,19 +7589,20 @@ static void atkA1_counterdamagecalculator(void)
     }
 }
 
-// a copy of atkA1 with the physical -> special field changes
-static void atkA2_mirrorcoatdamagecalculator(void)
+static void Cmd_mirrorcoatdamagecalculator(void) // a copy of Cmd with the physical -> special field changes
```
Comment (GriffinRichards) [inline src/battle_script_commands.c:7592]: Find and replace mistake here with `atkA1`. Also in general these comments should go above the function (I know this was synced from pokeemerald)

**Comment by AsparagusEduardo** [comment]
Ready for re-review.

**Pull request merged**